google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminClient.java

```diff
@@ -324,6 +324,34 @@ public void deleteTable(String tableId) {
     return transformToVoid(this.stub.deleteTableCallable().futureCall(request));
   }
 
+  /**
+   * Checks if the table specified by the tableId exists
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * if(client.exists("my-table")) {
+   *   "Table exists";
+   * }
+   * }</pre>
+   */
+  public boolean exists(String tableId) {
+    return awaitFuture(existsAsync(tableId)).booleanValue();
+  }
+
+  public ApiFuture<Boolean> existsAsync(final String tableId) {
+    return ApiFutures.transform(listTablesAsync(), new ApiFunction<List<TableName>, Boolean>() {
```

Comment:
I apologize, I didn't notice this earlier: the exists method should use GetTable using NAME_ONLY view. So it should be something like:

```java
GetTableRequest request = GetTableRequest.newBuilder().setName(tableName).setView(NAME_ONLY).build();

ApiFuture<Table> protoFuture = stub.getTableCallable().futureCall(request);
ApiFuture<Boolean> existsFuture = ApiFutures.transform(protoFuture, new ApiFunction<Table, Boolean>() {
  Boolean apply(Table ignored) { return true; }
})

return ApiFutures.catching(existsFuture, NotFoundException.class, new ApiFunction<NotFoundException, Boolean>() { Boolean apply(Throwable ignored) { return false});

```